lib/sqlite3.js

```diff
@@ -153,6 +153,77 @@ Database.prototype.removeAllListeners = function(type) {
     return val;
 };
 
+['all', 'each', 'get'].forEach(function (name) {
+    var allCallback = function(rowMode, callback) {
+        return function replacement(err, rows, fields) {
+            if (err) return callback(err);
+            if (!fields) return callback(null, rows);
+            var result = [];
+            var i, j, row;
+            if (rowMode == 'object') {
+                for (i = 0; i < rows.length; i++) {
+                    row = {};
+                    for (j = 0; j < fields.length; j++) {
+                        row[fields[j].name] = rows[i][j];
+                    }
+                    result.push(row);
+                }
+                callback(null, result);
+            }
+            else if (rowMode == 'nest') {
+                for (i = 0; i < rows.length; i++) {
+                    row = {};
+                    for (j = 0; j < fields.length; j++) {
+                        var table = row[fields[j].table];
+                        if (!table) table = row[fields[j].table] = {};
+                        table[fields[j].name] = rows[i][j];
+                    }
+                    result.push(row);
+                }
+                callback(null, result);
+            }
+            else {
+                callback(null, rows, fields);
+            }
+        };
+    };
+    var getCallback = function(rowMode, callback) {
+        var cb = allCallback(rowMode, function(err, rows, fields) {
+            callback(err, rows ? rows[0] : null, fields);
+        });
+        return function(err, row, fields) {
+            cb(err, [row], fields);
+        };
+    };
+    var wrapCallback = function(name, rowMode, callback) {
+        return name == 'all' ? allCallback(rowMode, callback) : getCallback(rowMode, callback);
+    };
+
+    var databaseMethodOrig = Database.prototype[name];
+    Database.prototype[name] = function () {
+        var args = Array.prototype.slice.call(arguments);
+        var sql = args.shift();
+        var rowMode = 'object';
+        if (sql == null) throw new Error('Unexpected sql null');
+        if (typeof sql == 'object') {
+            if (sql.rowMode) rowMode = sql.rowMode;
+            sql = sql.sql;
+        }
+        var callback = args.pop();
+        args.unshift(sql);
+        args.push(wrapCallback(name, rowMode, callback));
+        return databaseMethodOrig.apply(this, args);
+    };
+
+    var statementMethodOrig = Statement.prototype[name];
+    Statement.prototype[name] = function () {
+        var args = Array.prototype.slice.call(arguments);
+        var callback = args.pop();
+        args.push(callback.length < 3 ? wrapCallback(name, 'object', callback) : callback);
+        return statementMethodOrig.apply(this, args);
+    };
+});
```

Comment:
Override `all`, `each`, and `get` to be backward compatible. If array or nested rows is need, we can now: 

```js
db.all({ sql, rowMode: 'array' }, ...)
db.get({ sql, rowMode: 'nest' }, ...)
```